apis/datacatalog/v1alpha1/tagtemplate_reference.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ refsv1beta1.ExternalNormalizer = &TagTemplateRef{}
+
+// TagTemplateRef defines the resource reference to DataCatalogTagTemplate, which "External" field
+// holds the GCP identifier for the KRM object.
+type TagTemplateRef struct {
+	// A reference to an externally managed DataCatalogTagTemplate resource.
+	// Should be in the format "projects/{{projectID}}/locations/{{location}}/tagtemplates/{{tagtemplateID}}".
```

Comment:
nit: update field doc to use `tagTemplates`.

Reply:
done